src/wallet/wallet2.cpp

```diff
@@ -4397,6 +4673,9 @@ std::vector<wallet2::pending_tx> wallet2::create_transactions_2(std::vector<cryp
 
     // if we need to spend money and don't have any left, we fail
     if (unused_dust_indices.empty() && unused_transfers_indices.empty()) {
+      // ToDo:
+      //    the subaddress scheme tends to fail here when the requested amount is exactly the same as the subtotal of a subaddress
+      //    because the selection of subaddr_indices doesn't take the fee into account
```

Comment:
@moneromooo-monero Could you please help me with this problem? Currently, a problem like below can happen (which is quite likely to occur in practice):

```
[wallet/0 9xerQy]: balance detail
Currently selected account: [0] Primary account
Balance: 5.000000000000, unlocked balance: 5.000000000000
Balance per address:
        Address               Balance      Unlocked balance               Outputs   Label
       0 9xerQy        2.000000000000        2.000000000000                     1 Primary account
       1 BcVySV        3.000000000000        3.000000000000                     1        

[wallet/0 9xerQy]: transfer 9svHk1wHPo3ULf2AZykghzcye6sitaRE4MaDjPC6uanTHCynHjJHZaiAb922PojE1GexhhRt1LVf5DC43feyrRZMLXQr3mk 3
Error: Failed to find a way to create transactions. This is usually due to dust which is so small it cannot pay for itself in fees, or trying to send more money than the unlocked balance, or not leaving enough for fees
```
This happens because the requested amount 3 XMR is exactly the same as the subtotal balance for the subaddress index=1, so the wallet simply tries to construct a transaction using only funds received by it, but in the end the subaddress index=1 can't pay in full due to the fee required. In this case, the wallet should have chosen both indices 0 & 1 to collect unspent transfers for the input, but I couldn't implement this because I didn't know how to estimate the required fee.